### PR TITLE
Implement statx

### PR DESCRIPTION
**Pull request by FedorLap2006**
Implement `statx` function and `sys_statx` sysdep for Linux.

### CODE REVIEW

File: abis/linux/statx.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+#include <bits/types.h>
+
+struct statx_timestamp {
+	__mlibc_int64 tv_sec;
+	__mlibc_uint32 tv_nsec;
+	__mlibc_uint32 __padding;
+};
+
+struct statx {
```
Comment (no92) [inline abis/linux/statx.h:12]: Given that linux documents that the size of this struct is `0x100`, we should static_assert on it.

File: abis/linux/statx.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	__mlibc_uint32 stx_uid;
+	__mlibc_uint32 stx_gid;
+	__mlibc_uint16 stx_mode;
+	__mlibc_uint16 __padding;
+	__mlibc_uint64 stx_ino;
+	__mlibc_uint64 stx_size;
+	__mlibc_uint64 stx_blocks;
+	__mlibc_uint64 stx_attributes_mask;
+
+	struct statx_timestamp stx_atim;
```
Comment (no92) [inline abis/linux/statx.h:27]: You are missing `stx_btime`.

**Review by no92 — CHANGES_REQUESTED** [review]
Small mistakes, other than that LGTM.

**Comment by no92** [comment]
You should also reword the commit messages to align with the patches' contents and make sure that the commits are atomic.

**Review by qookei — APPROVED** [review]

**Review by Geertiebear — APPROVED** [review]

**Pull request merged**